src/ekat/io/ekat_yaml.cpp

```diff
@@ -1,43 +1,95 @@
+#include "ekat/util/ekat_meta_utils.hpp"
 #include "ekat/io/ekat_yaml.hpp"
 #include "ekat/ekat_assert.hpp"
 
 #include <yaml-cpp/yaml.h>
 
 #include <sstream>
 #include <fstream>
+#include <iomanip>
 
 namespace ekat {
 
+// These are the values we can parse.
+using values_t     = TypeList<bool,int,double,std::string>;
+
+// These are the vector<T> values we can store. char is to store bool.
+using seq_values_t = TypeList<char,int,double,std::string>;
+
 // =============================== READ ============================ //
 
 template<YAML::NodeType::value Type>
 void parse_node (const YAML::Node& node,
                  const std::string& key,
                  ParameterList& list);
 
-// Helpers
-bool is_bool (const std::string& s) {
+// Check if input string can be interpreted as given type
+template<typename T>
+bool is_type (const std::string& s);
+
+template<>
+bool is_type<bool> (const std::string& s) {
```

Comment:
Probably not needed, but if we used `strcasecmp` (or whatever C++ equivalent exists, if there is one), we could also allow all the interpolations between `TRUE` and `true`.